spring-integration-ftp/src/test/java/org/springframework/integration/ftp/config/FtpInboundChannelAdapterParserTests.java

```diff
@@ -100,6 +102,9 @@ public void testFtpInboundChannelAdapterComplete() throws Exception {
 		FtpInboundFileSynchronizingMessageSource inbound =
 			(FtpInboundFileSynchronizingMessageSource) TestUtils.getPropertyValue(ftpInbound, "source");
 
+		FileReadingMessageSource fileSource = (FileReadingMessageSource) TestUtils.getPropertyValue(inbound, "fileSource");
+		assertEquals(context.getBean("dirScanner"), fileSource.getScanner());
```

Comment:
That's definitely about `assertSame()` because we use dependency injection, therefore our `scanner` is indeed same as a bean.